p2p/peer_tracker.go

```diff
@@ -179,6 +206,17 @@ func (p *peerTracker) gc() {
 					delete(p.trackedPeers, id)
 				}
 			}
+
+			peerlist := make([]peer.AddrInfo, 0, len(p.trackedPeers))
+			for _, peer := range p.trackedPeers {
+				addrInfo := p.host.Network().Peerstore().PeerInfo(peer.peerID)
+				peerlist = append(peerlist, addrInfo)
+			}
+			err := p.peerstore.Put(context.Background(), peerlist)
```

Comment:
one thing to note here is that if we're passing in peerstore as an opt (and not setting it as a default), we'd have to check that the peerstore is not nil somewhere in order to do this, otherwise panic

There are two options here: 
1. Either pass some kind of noop peerstore by default so we don't have to do nil check (but i find this ugly) 
2. have a nil check

Reply:
I would have preferred the noop peerstore but the nil check is simpler